x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/expression/function/EsqlFunctionRegistry.java

```diff
@@ -295,6 +340,20 @@ public List<String> argDescriptions() {
         }
     }
 
+    private static DataType getTargetType(String[] names) {
+        List<DataType> types = new ArrayList<>();
+        for (String name : names) {
+            types.add(DataTypes.fromEs(name));
+        }
+        if (types.contains(KEYWORD) || types.contains(TEXT)) {
+            return DataTypes.NULL;
+        }
+
+        return types.stream()
+            .min((dt1, dt2) -> dataTypeCastingPriority.get(dt1).compareTo(dataTypeCastingPriority.get(dt2)))
+            .orElse(DataTypes.NULL);
```

Comment:
This probably shouldn't be `NULL` - if you leave off `orElse` then you'll get an Optional type.

Reply:
Let's try if UNSUPPORTED makes sense.